gradio/external.py

```diff
@@ -417,6 +419,35 @@ def get_spaces(model_name, api_key, alias, **kwargs):
         return get_spaces_blocks(model_name, config)
 
 
+async def get_pred_from_ws(
+    websocket: websockets.WebSocketClientProtocol, data: str
+) -> Dict[str, Any]:
+    completed = False
+    while not completed:
+        msg = await websocket.recv()
+        resp = json.loads(msg)
+        if resp["msg"] == "queue_full":
+            raise exceptions.Error("Queue is full! Please try again.")
```

Comment:
I think raising an error when the queue is full is the right thing to do since that's what the web client does but curious what other's thoughts are.

Reply:
Seems like a good option to me!